southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java

```diff
@@ -194,15 +202,28 @@ public void handle(final DeviceMappingConfigurationDTO configuration, final Map<
                     try {
                         bulk.dtos.addAll(handleRecord(configuration, globalState, record));
                     } catch (InvalidResourcePathException | ParserException | VariableNotFoundException e) {
-                        logger.error("Error parsing record with parser {}: {}", parserId, e.getMessage(), e);
+                        if (logErrors) {
+                            logger.error("Error parsing record with parser {}: {}", parserId, e.getMessage(), e);
+                        }
                     }
                 }
 
                 if (!bulk.dtos.isEmpty()) {
                     // Send all updates to the gateway thread at once
-                    dataUpdate.pushUpdate(bulk);
+                    Promise<?> pushUpdate = dataUpdate.pushUpdate(bulk);
+                    if (logErrors) {
+                        pushUpdate = pushUpdate.onFailure((t) -> logger
+                                .error("Error updating digital twin with parser {}: {}", parserId, t.getMessage(), t));
```

Comment:
I'd suggest doing a walk through of the failed updates if this is `FailedUpdatesException`